httpserver/http_connection.cpp

```diff
@@ -583,8 +591,31 @@ constexpr auto DB_POLL_INTERVAL = std::chrono::milliseconds(200);
 constexpr auto LONG_POLL_TIMEOUT = std::chrono::milliseconds(20000);
 constexpr uint32_t DB_POLL_LIMIT = LONG_POLL_TIMEOUT / DB_POLL_INTERVAL;
 
-/// TODO: get rid of argument copying
-void connection_t::poll_db(std::string pk, std::string last_hash) {
+template<typename T>
+void connection_t::respond_with_messages(const std::vector<T>& items) {
+
+    json res_body;
+    json messages = json::array();
+
+    for (const auto& item : items) {
+        json message;
+        message["hash"] = item.hash;
+        /// TODO: calculate expiration time once only?
+        message["expiration"] = item.timestamp + item.ttl;
```

Comment:
I think you can do ` message["expiration"] = item.expiration_timestamp;` here?

Reply:
I wish, but T is sometimes of type `message_t`, which doesn't have that field.

Reply:
Ah right because we are handing the message over in memory rather than checking the db again?
Hm

Reply:
Exactly. Would be nice not having to compute it twice, but this was the easiest way to make it work for now.